src/TiciTacaToeyGameEngine.ts

```diff
@@ -11,11 +11,44 @@ import {
   COMPLETED_GAME_STATUS,
   CalculateWinnerInputType,
   CalculateWinnerOutputType,
+  Player,
 } from "./model";
 import WebSocket = require("ws");
 import uniq from "lodash.uniq";
 
 const EMPTY_POSITION = "-";
+const DEFAULT_TIME = 30;
+
+class timer {
+  intervalId;
+  counter = DEFAULT_TIME;
+  // constructor() {}
+  start() {
+    this.intervalId = setInterval(() => {
+      this.counter = this.counter - 1;
+      console.log(this.counter);
+      if (this.counter === 0) clearInterval(this.intervalId);
+    }, 1000);
```

Comment:
We might want higher precision than 1 second, also we should move this out into a constant